mlfinlab/ensemble/sb_bagging.py

```diff
@@ -0,0 +1,662 @@
+"""
+Implementation of Sequentially Bootstrapped Bagging Classifier using sklearn's library as base class
+"""
+import numbers
+import itertools
+from warnings import warn
+from abc import ABCMeta, abstractmethod
+import pandas as pd
+import numpy as np
+
+from sklearn.tree import DecisionTreeClassifier, DecisionTreeRegressor
+from sklearn.metrics import accuracy_score, r2_score
+from sklearn.ensemble.bagging import BaseBagging, BaggingClassifier, BaggingRegressor
+from sklearn.base import ClassifierMixin, RegressorMixin
+from sklearn.ensemble.base import _partition_estimators
+from sklearn.utils.random import sample_without_replacement
+from sklearn.utils.validation import has_fit_parameter
+from sklearn.utils import check_random_state, indices_to_mask, check_array, check_consistent_length, check_X_y
+from sklearn.utils._joblib import Parallel, delayed
+
+from mlfinlab.sampling.bootstrapping import seq_bootstrap, get_ind_matrix
+
+MAX_INT = np.iinfo(np.int32).max
+
+
+# pylint: disable=too-many-ancestors
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-statements
+# pylint: disable=invalid-name
+# pylint: disable=protected-access
+# pylint: disable=len-as-condition
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=bad-super-call
+# pylint: disable=no-else-raise
+
+
+def _generate_indices_standard(random_state, bootstrap, n_population, n_samples):
```

Comment:
This method is only used to randomly generate features with replacement. I think we should rename the method appropriately or remove it entirely and just use `random_state.randint(0, n_features, max_features)` below. Leaving it as it is leads to confusion.

Reply:
If bootstrap_features is True, we generate features without replacement

Reply:
As far as I can see, the code does exactly the other way around. 
`
def _generate_indices_standard(random_state, bootstrap, n_population, n_samples):
    """Draw randomly sampled indices."""
    # Draw sample indices
    if bootstrap:
        indices = random_state.randint(0, n_population, n_samples)
    else:
        indices = sample_without_replacement(n_population, n_samples,
                                             random_state=random_state)

    return indices
`

I suggest refactoring this as you always call it with `bootstrap = True`

Reply:
You have 2 params: bootstrap and bootstrap features. Bootstrap features can be both True/False, it is Classifier param, so both parts of the code can be executed

Reply:
Indeed, you are right about that. Then maybe just rename the function so as to reflect that it is only used for random feature generation and not for samples.

Reply:
Sure